alcs/acs2/AbstractCondition.py

```diff
@@ -0,0 +1,40 @@
+from alcs import Perception
+from alcs.acs2 import default_configuration
+
+
+class AbstractCondition(list):
+    def __init__(self, seq=(), cfg=None):
+        self.cfg = cfg or default_configuration
+        if not seq:
+            list.__init__(
+                self,
+                [self.cfg.classifier_wildcard] * self.cfg.classifier_length)
+        else:
+            list.__init__(self, seq)
+            if len(self) != self.cfg.classifier_length:
+                raise ValueError('Illegal length of effect string')
+
+    def __setitem__(self, idx, value):
+        if not isinstance(value, str):
+            raise TypeError('Effect should be composed of string objects')
+
+        super(AbstractCondition, self).__setitem__(idx, value)
+
+    def __repr__(self):
+        return ''.join(map(str, self))
+
+    def is_specializable(self, p0: Perception, p1: Perception) -> bool:
```

Comment:
AFAIK it applies only to `Effect` class and should not be generalized.

Reply:
True. Will push it back down to `Effect`.